src/lib/zf/attr.c

```diff
@@ -286,6 +287,14 @@ int zf_attr_set_str(struct zf_attr* attr, const char* name, const char* val)
       *p = strdup(val);
       if( *p == NULL )
         return -ENOMEM;
+      if( strcmp(name, "log_file")==0 ) {
+        int rc = zf_log_redirect(attr->log_file);
+        if( rc < 0 ) {
+          zf_log_stack_err(NO_STACK, "%s: Failed to redirect logging: %s\n",
+                          __func__, strerror(-rc));
+          /* Failure here is non-fatal. */
```

Comment:
Interesting, it seems a safer approach. Are you suggesting this should also be the case in zf_init()? that is terminating the initialization with an error, EPERM for example. 

https://github.com/andresa-xilinx/tcpdirect/blob/173a39aec69fc0d83c87af3837478700c3a4c64a/src/lib/zf/zf.c#L32

Reply:
The logging output from `zf_init()` comes from environment rather than a direct API call to configure.  You could make a case for giving an error from `zf_init()` due to bad logging related env vars but I wouldn't change current behaviour now.

I do think this case where an API call with the single purpose of reconfiguring logging should return an error on failure though.  I wouldn't make it EPERM though as there are many reasons (ref `man 2 open`) for not being able to open file.  I would suggest more generic EINVAL. I advise getting a second opinion too.

Reply:
> there are many reasons (ref man 2 open) for not being able to open file.

Wouldn't that suggest you should pass through the errno? Squashing errors to EINVAL is sometimes necessary but never nice!